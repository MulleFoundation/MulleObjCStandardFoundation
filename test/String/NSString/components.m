//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif



static void  test( NSString *s, NSString *seperator)
{
   NSArray    *array;
   NSString   *component;

   array = [s componentsSeparatedByString:seperator];
   printf( "\"%s\" separated by \"%s\":",
            [s UTF8String],
            [seperator UTF8String]);

   for( component in array)
      printf( " \"%s\"", [component UTF8String]);
   printf( "\n");
}

int main( int argc, const char * argv[])
{
   test( @"", @"");

NS_DURING
   test( @"0,1,2,3", nil);
NS_HANDLER
   printf( "%s\n", [[localException name] UTF8String]);
NS_ENDHANDLER

   test( @"0,1,2,3", @",");
   test( @"0,1,2,3", @",,");
   test( @"0,1,,2,,,,3,,,,4", @",");
   test( @"0,1,,2,,,,3,,,,4", @",,");

   test( @",", @",");
   test( @"0,", @",");
   test( @",0,", @",");
   test( @"0,,", @",");
   test( @",0", @",");
   test( @",,0", @",");
   test( @",,0,,", @",");
   test( @",,0,0,,", @",");

   return( 0);
}
