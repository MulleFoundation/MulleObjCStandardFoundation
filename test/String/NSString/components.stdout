"" seperated by "": ""
NSInvalidArgumentException
"0,1,2,3" seperated by ",": "0" "1" "2" "3"
"0,1,2,3" seperated by ",,": "0,1,2,3"
"0,1,,2,,,,3,,,,4" seperated by ",": "0" "1" "" "2" "" "" "" "3" "" "" "" "4"
"0,1,,2,,,,3,,,,4" seperated by ",,": "0,1" "2" "" "3" "" "4"
"," seperated by ",": "" ""
"0," seperated by ",": "0" ""
"0,," seperated by ",": "0" "" ""
",0" seperated by ",": "" "0"
",,0" seperated by ",": "" "" "0"
",,0,," seperated by ",": "" "" "0" "" ""
",,0,0,," seperated by ",": "" "" "0" "0" "" ""
